<?xml version="1.0"?>
<launch>

    <!--
        Change the value for camera_topic for the input subscriber topic
    -->
    <arg name="camera_topic" default="/kinect2/qhd/image_color" />

    <arg name="model" default="mobilenet_thin" />
    <arg name="resolution" default="640x480" />
    <arg name="gpuid" default="0" />

    <node name="pose_estimator" pkg="Gesture_Recognition_ROS" type="initialize_tfpose.py" output="screen" required="true">
        <!--
            Parameters for tf-pose-estimation
        -->
        <env name="CUDA_VISIBLE_DEVICES" value="$(arg gpuid)" />
        <param name="camera" value="$(arg camera_topic)"/>
        <param name="model" value="$(arg model)" />
        <param name="resolution" value="$(arg resolution)" />

        <!--
            Parameters for inference layer
        -->
        <param name="frame_rate"    value="4"/>
        <param name="use_arm"       value="False" />
        <param name="use_angles"    value="False" />
        <param name="bad_data_flag" value="False" />
        <param name="debug"         value="False" />

    </node>

    <node name="pose_estimator_vis" pkg="Gesture_Recognition_ROS" type="visualization.py" output="screen" required="true">
        <param name="camera" value="$(arg camera_topic)"/>
        <param name="pose" value="/pose_estimator/pose" />
    </node>
<!--
    <arg name="node_start_delay" default="10.0" />
    <node name="LSTM_Inference" pkg="Gesture_Recognition_ROS" type="lstm_inference.py" output="screen" required="true" launch-prefix="bash -c 'sleep $(arg node_start_delay); $0 $@' ">
    </node>
-->
</launch>
